### ISSUE TITLE
fgdc spatial reference information

### ISSUE THREAD

**Issue by stansmith907** [adiwg/mdTranslator#127]
Spatial reference information (section 4) in FGDC does not align with our implementation of ISO and mdJson.  FGDC does not provide a means to declare spatial references via CRS or EPSG numbers.  FGDC only provides for numerical definitions of projections.  This can also be done in ISO, however this was a section we skipped in our implementation.  

To properly translate FGDC spatial reverence to ISO 19115:2003 or -2 we would need to add the missing classes MD_ProjectionParameters, MD_EllipsoidParameters, MD_ObliqueLineAzimuth, and MD_ObliqueLineProject.  ISO 19115-2 does provide an *identifier* to declare EPSG numbers which we support.

To properly translate FGDC to ISO 19115-1 we would need to additionally implement ISO 19111.  ISO 19115-1 only accepts EPSG numbers referenced by an *identifier*.  This was the primary reason we did not include the above classes in our implementation.  

In summary, we cannot get this information into 19115-2 without a lot of work and cannot get it into 19115-1 without generating a second ISO record in 19111 which would be referenced in 19115-1 via an identifier, even more work.  So, I am skipping section 4 of FGDC until we discuss this.

**Comment by dwalt** [comment]
Consider a parameter input identifying epsg no for spatial reference
system. There are data quality elements (lat/long resolution and coordinate
units) in FGDC spatial reference that should be mapped to mdJSON elements.
If using custom projection parameters, then that might be the 2% of the
problem we don't support at this time.

**Comment by stansmith907** [comment]
FGDC defines 22 spatial projections.

 - [x] convert to WKT and report in ISO 19115-2 as *code* in referenceSystemInfo
 - [ ] convert to ISO 19115-2 classes MD_ProjectionParameters, MD_EllipsoidParameters, MD_ObliqueLineAzimuth, and MD_ObliqueLineProject; extend mdJson/mdTranslator models
 - [ ] convert to ISO 19111 and convert these in to ISO 19115-2 writer; add ISO 19111 writer; extend mdJson/mdTranslator models
 - [ ] ignore this section

Not sure which, if any, of the above choices will facilitate proper FGDC writer output.

**Comment by stansmith907** [comment]
I stopped work on converting FGDC spatial representation to WKT for the following reasons:
 * WKT does not cover the entire FGDC section
 * difficult to convert to and from FGDC
 * does not support output to ISO 19115-2
 * does not move us closer to support for ISO 19111

I added ISO 19115-2 classes to the mdJson/mdTranslator and will map FGDC to those classes.  This is working for the 22 planar projections and possibly the 4 local coordinate systems.  However new classes will need to be added for describing geodetic models. 

 FGDC *geographic* definition is "the quantities of latitude and longitude which define the position of a point on the Earth's surface with respect to a reference spheroid".  This is representative of our *spatialResolution" and moved to issue

**Comment by stansmith907** [comment]
Current status:
* I do not have ISO 19111 documentation, I asked Dave Blodgett for assistance but he could not come up with anything.  Will ask Tim Kern for assistance.
* Some (or all) of the model is included in the ISO 19115:2003 object model.  But it seems the XSD was never completed so adding the objects causes the validation to fail.
* The ISO 19111 model seems to be detached from ISO 19115-3.  I can download the ISO 19111 XSD from the -3 XSD folders, I just don't know how to link this with 19115-3 yet.  Will ask Ted for insight.
